crates/runtime/src/instance/allocator/pooling.rs

```diff
@@ -501,6 +544,7 @@ impl InstancePool {
         self.free_list.lock().unwrap().push(index);
```

Comment:
Ah, yes. I went looking for `info` and realized this was part of the lazy-anyfunc changes that have been split off. But that just proves the point more: it's entirely unobvious in the other PR that this would need to change, and a review of that PR could have easily missed it.

I think the invariant of keeping `Instance`s initialized-but-empty and "hot-patching" the fields was probably intended to keep the `PrimaryMap`s from being freed/allocated on each instantiation -- at least that's one benefit I see -- but I agree that the risk here is probably greater than the reward, and actually aligns with some of the points above re: instantiation from scratch vs. erasing state to reuse, just on a micro scale.

I went ahead and refactored so that the instances remain uninitialized memory until instantiation, and are `drop_in_place()`'d on termination. Good catch!